sklearn/metrics/_classification.py

```diff
@@ -1483,15 +1483,23 @@ def precision_recall_fscore_support(y_true, y_pred, *, beta=1.0, labels=None,
     if average == 'weighted':
         weights = true_sum
         if weights.sum() == 0:
-            zero_division_value = 0.0 if zero_division in ["warn", 0] else 1.0
```

Comment:
Please explain why these changes were necessary. Thanks

Reply:
Hi @jnothman actually this change is also for the same reason. The `zero_division_value` is a float and will not have `.item()` function 

I my recent changes I reverted back the `float` change for each `i.item()` as it was making `support` a `float` as well which was causing a test failure.

Reply:
So what you're saying is that this is fixing the problem underlying the classification_report error: it assumed all the results of precision, recall, etc, were numpy types, but this was not the case if `zero_division` was used?

Reply:
@jnothman yes that was exactly the case.